apps/extension/src/ui/apps/dashboard/routes/Portfolio/AccountContextMenu.tsx

```diff
@@ -0,0 +1,153 @@
+  const { account: selectedAccount } = useSelectedAccount()
+  const account =
+    (address === null
+      ? // if address prop is null, set account to null
+        null
+      : address === undefined
+      ? // if address prop is undefined, set account to selectedAccount
+        selectedAccount
+      : // if address prop is a string, set account to propsAccount
+        propsAccount) ??
+    // make sure account is either an `Account` or undefined
+    undefined
+
+  const navigate = useNavigate()
+  const { genericEvent } = useAnalytics()
+
+  const goToManageAccounts = useCallback(() => navigate("/settings/accounts"), [navigate])
+
+  const { canToggleIsPortfolio, toggleIsPortfolio, toggleLabel } =
+    useAccountToggleIsPortfolio(account)
+
+  // TODO: These modal providers used to be used in multiple places,
+  // hence the hectic API we've got going on here.
+  // We should clean them up to just support this one component's use-case.
+  const { open: openCopyAddressModal } = useCopyAddressModal()
+  const canCopyAddress = !!account
+  const copyAddress = useCallback(() => {
+    if (!account) return
+    genericEvent("open copy address", { from: analyticsFrom })
+    openCopyAddressModal({ mode: "copy", address: account.address })
+  }, [account, analyticsFrom, genericEvent, openCopyAddressModal])
+
+  const showTxHistory = useIsFeatureEnabled("LINK_TX_HISTORY")
+  const openTxHistory = useCallback(() => {
+    genericEvent("open web app tx history", { from: analyticsFrom })
+    window.open(getTransactionHistoryUrl(account?.address), "_blank")
+  }, [account?.address, analyticsFrom, genericEvent])
+
+  const { open: _openAccountRenameModal } = useAccountRenameModal()
+  const canRename = !!account
+  const openAccountRenameModal = useCallback(
+    () => _openAccountRenameModal(account),
+    [_openAccountRenameModal, account]
+  )
+
+  const { canExportAccountFunc, open: _openAccountExportModal } = useAccountExportModal()
+  const canExport = useMemo(() => canExportAccountFunc(account), [account, canExportAccountFunc])
+  const openAccountExportModal = useCallback(
+    () => _openAccountExportModal(account),
+    [_openAccountExportModal, account]
+  )
+
+  const { canExportAccountFunc: canExportAccountPkFunc, open: _openAccountExportPkModal } =
+    useAccountExportPrivateKeyModal()
+  const canExportPk = useMemo(
+    () => canExportAccountPkFunc(account),
+    [account, canExportAccountPkFunc]
+  )
+  const openAccountExportPkModal = useCallback(
+    () => _openAccountExportPkModal(account),
+    [_openAccountExportPkModal, account]
+  )
+
+  const { canRemoveFunc, open: _openAccountRemoveModal } = useAccountRemoveModal()
+  const canRemove = useMemo(() => canRemoveFunc(account), [account, canRemoveFunc])
+  const openAccountRemoveModal = useCallback(
+    () => _openAccountRemoveModal(account),
+    [_openAccountRemoveModal, account]
+  )
+
+  return (
+    <ContextMenu placement="bottom-end">
```

Comment:
as button is on the left side of the screen, it feels weird if menu opens towards the left

```suggestion
    <ContextMenu placement="bottom-start">
```